File: tests/integration/scripts/filesystem_test.js
#!/usr/bin/env node
/**
 * Filesystem operations test script for agent-jail integration testing.
 * Creates, reads, modifies, and deletes files to verify filesystem event capture.
 */
const fs = require('fs');
const path = require('path');
const os = require('os');

async function main() {
    console.log("=== Node.js Filesystem Test ===");

    // Create a temp directory
    const testDir = fs.mkdtempSync(path.join(os.tmpdir(), 'jail_test_node_'));
    console.log(`Created temp dir: ${testDir}`);

    // Create a file (sync)
    const testFile = path.join(testDir, 'test_file.txt');
    fs.writeFileSync(testFile, 'Hello from Node.js!\n');
    console.log(`Created file: ${testFile}`);

    // Read the file
    const content = fs.readFileSync(testFile, 'utf-8');
    console.log(`Read content: ${content.trim()}`);

    // Append to file
    fs.appendFileSync(testFile, 'Second line\n');
    console.log('Appended to file');

    // Create subdirectory
    const subDir = path.join(testDir, 'subdir');
    fs.mkdirSync(subDir);
    console.log(`Created subdir: ${subDir}`);

    // Create file in subdirectory
    const nestedFile = path.join(subDir, 'nested.json');
    fs.writeFileSync(nestedFile, JSON.stringify({ status: 'success', nested: true }));
    console.log(`Created nested file: ${nestedFile}`);

    // Async file operations
    const asyncFile = path.join(testDir, 'async_test.txt');
    await fs.promises.writeFile(asyncFile, 'Async write!\n');
    console.log(`Created async file: ${asyncFile}`);

    const asyncContent = await fs.promises.readFile(asyncFile, 'utf-8');
    console.log(`Async read: ${asyncContent.trim()}`);

    // List directory
    const files = fs.readdirSync(testDir);
    console.log(`Directory contents: ${JSON.stringify(files)}`);

    // File stats
    const stats = fs.statSync(testFile);
    console.log(`File stats: size=${stats.size}, isFile=${stats.isFile()}`);

    // Delete files
    fs.unlinkSync(nestedFile);
    console.log(`Deleted: ${nestedFile}`);

    fs.rmdirSync(subDir);
    console.log(`Deleted dir: ${subDir}`);

    fs.unlinkSync(asyncFile);
    fs.unlinkSync(testFile);
    console.log(`Deleted test files`);

    fs.rmdirSync(testDir);
    console.log(`Deleted temp dir: ${testDir}`);

    console.log("=== Filesystem Test Complete ===");
    return 0;
}

main().then(process.exit).catch(err => {
    console.error(err);
    process.exit(1);
});

File: tests/integration/scripts/filesystem_test.py
#!/usr/bin/env python3
"""
Filesystem operations test script for agent-jail integration testing.
Creates, reads, modifies, and deletes files to verify filesystem event capture.
"""
import os
import sys
import tempfile
import json

def main():
    print("=== Python Filesystem Test ===")

    # Create a temp directory
    test_dir = tempfile.mkdtemp(prefix="jail_test_")
    print(f"Created temp dir: {test_dir}")

    # Create a file
    test_file = os.path.join(test_dir, "test_file.txt")
    with open(test_file, 'w') as f:
        f.write("Hello from Python!\n")
    print(f"Created file: {test_file}")

    # Read the file
    with open(test_file, 'r') as f:
        content = f.read()
    print(f"Read content: {content.strip()}")

    # Append to file
    with open(test_file, 'a') as f:
        f.write("Second line\n")
    print("Appended to file")

    # Create a subdirectory
    sub_dir = os.path.join(test_dir, "subdir")
    os.makedirs(sub_dir)
    print(f"Created subdir: {sub_dir}")

    # Create file in subdirectory
    nested_file = os.path.join(sub_dir, "nested.json")
    with open(nested_file, 'w') as f:
        json.dump({"status": "success", "nested": True}, f)
    print(f"Created nested file: {nested_file}")

    # List directory
    files = os.listdir(test_dir)
    print(f"Directory contents: {files}")

    # Delete files
    os.remove(nested_file)
    print(f"Deleted: {nested_file}")

    os.rmdir(sub_dir)
    print(f"Deleted dir: {sub_dir}")

    os.remove(test_file)
    print(f"Deleted: {test_file}")

    os.rmdir(test_dir)
    print(f"Deleted temp dir: {test_dir}")

    print("=== Filesystem Test Complete ===")
    return 0

if __name__ == "__main__":
    sys.exit(main())
